activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -303,6 +303,13 @@ def +(other)
         result = utc + other
         result.in_time_zone(time_zone)
       end
+    rescue TypeError
```

Comment:
Having fairly generic rescues like this can be confusing if 

Could we make this rescue only capture only exceptions from the `utc + other` line? That way we're only going to rescue AT MOST exceptions we would have in previous versions of Rails.